files/default/slurm/pcluster_slurm_config_generator.py

```diff
@@ -90,8 +90,17 @@ def _get_head_node_config():
 def _get_head_node_private_ip():
     """Get head node private ip from EC2 metadata."""
     try:
+        token = subprocess.run(  # nosec nosemgrep
```

Comment:
Yes, we probably don't want to share a helper function, but we can at least put this `subprocess.run` call that uses all the options into a helper function within this script, so we don't have to write everything twice, let's at least do this.

Nice to have: We can also use the `requests` based way of retrieving metadata as well, since that is more pythonic than using a `subprocess` call

Reply:
Right! Changed it used requests and add `_get_metadata` function in the file